src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -1037,5 +1034,42 @@ private static IEnumerable<IMSBuildItem> GetDistinctItemsOrEmpty(IMSBuildProject
         {
             return project.GetItems(itemName)?.Distinct(ProjectItemInstanceEvaluatedIncludeComparer.Instance) ?? Enumerable.Empty<IMSBuildItem>();
         }
+
+        /// <summary>
+        /// Logs an error from the specified exception.
+        /// </summary>
+        /// <param name="exception">The <see cref="Exception" /> with details to be logged.</param>
+        private void LogErrorFromException(Exception exception)
```

Comment:
It'd be great if we can add some tests here.

Reply:
I've added a functional test